rest/multipart.go

```diff
@@ -46,12 +47,17 @@ func ReadJSONFromMIME(headers http.Header, input io.Reader, into interface{}) er
 		return base.HTTPErrorf(http.StatusUnsupportedMediaType, "Unsupported Content-Encoding; use gzip")
 	}
 
+	defer func() {
+		_ = input.Close()
```

Comment:
As discussed, can move this down to avoid the defer, by returning the decoder.Decode error on line 61